mathesar_ui/src/component-library/common/types/ArtificialEvents.ts

```diff
@@ -0,0 +1,4 @@
+export interface ArtificialEvents<T> {
+  artificialInput: T;
+  artificialChange: T;
+}
```

Comment:
Yeah, I also really don't like the mess of `input` and `artificialInput`. I wrestled with this challenge quite a bit in this PR, trying a number of different approaches to debounce a dynamic input. I was hoping to find a design that wouldn't require modifying _all_ of our input components. I think this topic deserves a discussion via a sync call so that we can establish a good pattern to use throughout the component library. My opinion on the best design has been continuing to evolve throughout my work on this PR, but where I currently stand is that I'd like to use _props_ (instead of events) to handle imperative upwards data flow from all our input components. We've debated this before, but I think it's worth revisiting in the context of this specific use-case.

Reply:
Okay. I think we can have a frontend sync call for this during the cooldown. I'll leave it to you to schedule one.